### PR TITLE
Supplementary reason codes

### LINKED ISSUE #139: Reason codes should be consistent
I implemented `incident.reason_code` as an AR model but `incident.second_reason_code` as an array of constants. This is dumb and inconsistent for no reason other than that I was rushing.

Should go back and make a `SupplementaryReasonCode` or something.

### PR DESCRIPTION
**Pull request by dfaulken**
Closes #139.

1. Merge / deploy / migrate
2. `rake supplementary_reason_codes:create && rake supplementary_reason_codes:migrate`
3. Create a migration removing the disused column.

### CODE REVIEW

File: app/models/reason_code.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 class ReasonCode < ApplicationRecord
   has_many :incidents
-  validates :identifier, :description, presence: true, uniqueness: true
+  validates :identifier, :description, presence: true
+  validates :identifier, uniqueness: true # description isn't unique, apparently
```
Comment (mboneil10) [inline app/models/reason_code.rb]: What do you mean by apparently? This was requested? Or something from Hastus?

Comment (dfaulken) [reply app/models/reason_code.rb]: I'd have to go back and look at the history of where these came from to be 100% sure, but I'm pretty sure it's a list configured by SATCO.

Comment (mboneil10) [reply app/models/reason_code.rb]: Alright, cool

File: spec/system/staff/editing_incident_details_spec.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@
   let(:incident) { incident_in_divisions staff.divisions }
   it 'allows editing reason codes' do
     create :reason_code, identifier: 'A-1', description: 'Falling bananas'
+    create :supplementary_reason_code, identifier: 'a-8', description: 'Miscellaneous'
```
Comment (mboneil10) [inline spec/system/staff/editing_incident_details_spec.rb]: Why can't you use the `identifier` and `description` defined from the factory?

File: lib/tasks/supplementary_reason_codes.rake [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                                                 description: description
+      unless new_code.full_label == code
+        fail 'You dun goofed'
+      end
+    end
+  end
+
+  task migrate: :environment do
+  # Example invocation: rails supplementary_reason_codes:migrate
+    incidents = Incident.where.not second_reason_code: nil
```
Comment (mboneil10) [inline lib/tasks/supplementary_reason_codes.rake]: I've never seen `.not` before. That's great!

File: lib/tasks/supplementary_reason_codes.rake [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    incidents.each do |incident|
+      identifier = incident.second_reason_code.split(': ').first
+      code = SupplementaryReasonCode.find_by! identifier: identifier
+      incident.update supplementary_reason_code: code
```
Comment (mboneil10) [inline lib/tasks/supplementary_reason_codes.rake]: Will this `update` ever fail?

File: spec/system/staff/editing_incident_details_spec.rb (hunk 2 above)
Comment (dfaulken) [reply spec/system/staff/editing_incident_details_spec.rb]: I can, just chose to modify as little of the existing test as possible

File: lib/tasks/supplementary_reason_codes.rake (hunk 4 above)
Comment (dfaulken) [reply lib/tasks/supplementary_reason_codes.rake]: Sure! But only for reasons that were already a problem before we got here (changing validity standards, etc.)

File: lib/tasks/supplementary_reason_codes.rake (hunk 3 above)
Comment (dfaulken) [reply lib/tasks/supplementary_reason_codes.rake]: I think so too!

File: spec/system/staff/editing_incident_details_spec.rb (hunk 2 above)
Comment (mboneil10) [reply spec/system/staff/editing_incident_details_spec.rb]: Ah, got it.

File: lib/tasks/supplementary_reason_codes.rake (hunk 4 above)
Comment (mboneil10) [reply lib/tasks/supplementary_reason_codes.rake]: Great.

File: app/controllers/incidents_controller.rb [hunk 5]
```diff
@@ -131,6 +133,8 @@ def unclaimed
   end
 
   def update
+    @reason_codes = ReasonCode.order(:identifier)
+    @supplementary_reason_codes = SupplementaryReasonCode.order(:identifier)
```
Comment (Anbranin) [inline app/controllers/incidents_controller.rb]: Why add these instance variables right here?

Comment (dfaulken) [reply app/controllers/incidents_controller.rb]: As opposed to a `before_action`, you're saying? Or are you questioning something else?

Comment (Anbranin) [reply app/controllers/incidents_controller.rb]: Just not sure if I'm missing something big--update actions don't usually respond to get requests so where are these used?

Comment (dfaulken) [reply app/controllers/incidents_controller.rb]: The `else` renders a view that needs them.

Comment (Anbranin) [reply app/controllers/incidents_controller.rb]: OMG I knew I was missing something. 👯‍♂️

**Pull request merged**